android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxgl/views/MapView.java

```diff
@@ -428,6 +429,25 @@ public void removeAnnotations() {
         return Collections.unmodifiableList(mAnnotations);
     }
 
+    public List<Annotation> getAnnotationsInBounds(BoundingBox bbox) {
+        List<Annotation> annotations = new ArrayList<>();
+        long[] ids = mNativeMapView.getAnnotationsInBounds(bbox);
+        List<Long> idsList = new ArrayList<>();
+        for(int i = 0; i < ids.length; i++) {
+            idsList.add(new Long(ids[i]));
+        }
+        for(int i = 0; i < mAnnotations.size(); i++) {
+            Annotation annotation = mAnnotations.get(i);
+            if (annotation instanceof Marker && idsList.contains(annotation.getId())) {
+                annotations.add(annotation);
+            }
+        }
```

Comment:
Since `mAnnotations` is defined using Generics `List<Annotation>` it can be iterated easier and safer via:

```java
for (Annotation annotation : mAnnotations) {
...
}
```